### PR TITLE
Use template literals for dsn.js

### PR DESCRIPTION
**Pull request by rhedshi**
updates #2129 

Changes proposed in this pull request:
- Use template literals for dsn.js

Checklist:
- [ ] docs updated
- [ ] tests updated
- [ ] [Changes](https://github.com/haraka/Haraka/blob/master/Changes.md) updated

### CODE REVIEW

File: dsn.js [hunk 1]
```diff
@@ -85,11 +85,11 @@ function DSN (code, msg, def, subject, detail) {
         this.reply = [];
         let m;
         while ((m = this.msg.shift())) {
-            this.reply.push([this.cls, this.sub, this.det].join('.') + ' ' + m);
+            this.reply.push(`${[this.cls, this.sub, this.det].join('.')} ${m}`);
         }
     }
     else {
-        this.reply = [this.cls, this.sub, this.det].join('.') + ' ' + (this.msg || this.default_msg);
+        this.reply = `${[this.cls, this.sub, this.det].join('.')} ${(this.msg || this.default_msg)}`;
```
Comment (msimerson) [inline dsn.js]: lets get rid of the join('.') here as well and just use template literals fully. Eg: 

```js
this.reply = `${this.cls}.${this.sub}.${this.det} ${(this.msg || this.default_msg)}`;
```

**Comment by rhedshi** [comment]
Uh oh. I think I made a bad merge...

**Comment by msimerson** [comment]
Yep, you bumped into another merge that added es6 classes.   Since the changes in this PR are so small it might be easier just to create a new branch from master in manually applay them in a new PR.

**Comment by rhedshi** [comment]
Updated here in #2265

**Pull request closed without merging**